src/Microsoft.AspNet.Mvc.Core/ModelBinding/ByteArrayModelBinder.cs

```diff
@@ -13,19 +13,22 @@ namespace Microsoft.AspNet.Mvc.ModelBinding
     public class ByteArrayModelBinder : IModelBinder
     {
         /// <inheritdoc />
-        public async Task<ModelBindingResult> BindModelAsync([NotNull] ModelBindingContext bindingContext)
+        public Task<ModelBindingResult> BindModelAsync([NotNull] ModelBindingContext bindingContext)
         {
             // Check if this binder applies.
             if (bindingContext.ModelType != typeof(byte[]))
             {
-                return null;
+                return Task.FromResult<ModelBindingResult>(null);
             }
 
             // Check for missing data case 1: There was no <input ... /> element containing this data.
-            var valueProviderResult = await bindingContext.ValueProvider.GetValueAsync(bindingContext.ModelName);
+            var valueProviderResult = bindingContext.ValueProvider.GetValue(bindingContext.ModelName);
             if (valueProviderResult == ValueProviderResult.None)
             {
-                return new ModelBindingResult(model: null, key: bindingContext.ModelName, isModelSet: false);
+                return Task.FromResult(new ModelBindingResult(
+                    model: null,
+                    key: bindingContext.ModelName,
+                    isModelSet: false));
```

Comment:
Wondering about avoiding repeated `Task.FromResult(...)` in binders that run synchronously. The following might be cleaner...
``` c#
public Task<ModelBindingResult> BindModelAsync([NotNull] ModelBindingContext bindingContext)
{
    return Task.FromResult(BindModel(bindingContext));
}

private ModelBindingResult BindModel(ModelBindingContext bindingContext)
{
    // the old BindModelAsync() content, less some awaiting
}
```